homeassistant/helpers/discovery_flow.py

```diff
@@ -47,6 +47,9 @@ def _async_init_flow(
     if hass.config_entries.flow.async_has_matching_flow(domain, context, data):
         return None
 
+    if hass.is_stopping:
```

Comment:
let's move this up, it's a tiny check and faster than has_matching_flow.

Reply:
a58d6f8ad731b7ee36cf98b6c2633472384c07ba